### DISCUSSION TITLE
List Users Method only returns an array with a fixed length.

### DISCUSSION THREAD

**Discussion by cwancy** [supabase/supabase#8865] [Questions] [answered]
Hello everyone,

My team use Supabase for one of our production applications. We have run into an issue where the response from the `const users = await supabase.auth.api.listUsers()` will return only an array of 50 users. We have a lot more users than that, and any users that are not included in the initial array of 50 users, are excluded in the response from the API call. 

An example of this is below:

This first image is the user list from the Supabase authentication dashboard. I have circled it in red, that we have 56 users in total. 

Now onto the second image, I am just console logging the `const users = await supabase.auth.api.listUsers()` method. However, only 50 users are listed in the response. 

I have verified that only 50 or less will come through at a time, as I deleted one user, and the users that were listed in the response were different, but it was still only 50. 

![56results](https://user-images.githubusercontent.com/60968585/189234912-dbbea424-1b7f-4536-9817-12750ee313e4.png)

![consolelogsupabase](https://user-images.githubusercontent.com/60968585/189235196-694c3fa5-18d9-4648-a628-5639f3229889.png)

I have checked through all of the Supabase logs, and I have tried to debug the issue client side - but I get nothing. I cannot find any resources online about this issue either.

Has anyone seen this before? 

Thanks

**Comment by GaryAustin1** [comment, marked as answer]
gotrue has pagination capabilities, but apparantly gotrue.js does not use them.
https://github.com/supabase/gotrue/blob/master/api/pagination.go
The default is set to 50... `const defaultPerPage = 50` in the above file.

I've not seen anyone discuss using the pagination with a fetch command, so not sure if it really works or not, but assume it does.

Probably easier to just write your own security definer function and get what you need from auth.users() with sql.  Just make sure to check for service_role access as part of the function.

**Reply by cwancy** [reply]
Thanks, apologies for late reply. Creating a function would be the most ideal way as stated above.
